Assets/Plugins/Source/Core/Utility/JsonUtility.cs

```diff
@@ -22,21 +22,106 @@
 
 namespace PlayEveryWare.EpicOnlineServices.Utility
 {
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+    using System;
+    using UnityEngine;
+
     public static class JsonUtility
     {
+        /// <summary>
+        /// Attempts to parse the given JSON string. If problems are encountered, an error will be logged and the exception will continue.
+        /// </summary>
+        /// <param name="json">The JSON string to validate.</param>
+        private static void ValidateJson(string json)
+        {
+            try
+            {
+                JToken.Parse(json);
+            }
+            catch (JsonReaderException ex)
+            {
+                Debug.LogError($"Invalid JSON: {ex.Message}");
+#if UNITY_EDITOR
+                throw;
+#endif
+            }
+            catch (Exception ex)
+            {
+                Debug.LogError($"An error occurred: {ex.Message}");
+#if UNITY_EDITOR
+                throw;
+#endif
```

Comment:
If the JSON is invalid, we want to throw an exception if we are in the editor. However, if we are running outside the editor, then we want to fail more silently.